src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -2049,10 +2057,82 @@ WaitForMiliseconds(long timeout)
  * replication.
  */
 static XLogRecPtr
-GetSubscriptionPosition(MultiConnection *connection, Bitmapset *tableOwnerIds)
+GetSubscriptionPosition(MultiConnection *connection, Bitmapset *tableOwnerIds,
+						char *operationPrefix)
 {
-	char *subscriptionValueList = ShardMoveSubscriptionNamesValueList(tableOwnerIds);
+	char *subscriptionValueList = ShardSubscriptionNamesValueList(tableOwnerIds,
+																  operationPrefix);
 	return GetRemoteLSN(connection, psprintf(
 							"SELECT min(latest_end_lsn) FROM pg_stat_subscription "
 							"WHERE subname IN %s", subscriptionValueList));
 }
+
+
+/*TODO(saawasek):Refactor this for ShardMove too.*/
+void
+CreateShardSubscription(MultiConnection *connection, char *sourceNodeName,
+						int sourceNodePort, char *userName, char *databaseName,
+						char *publicationName, char *slotName,
+						Oid ownerId)
+{
+	StringInfo createSubscriptionCommand = makeStringInfo();
+	StringInfo conninfo = makeStringInfo();
+
+	/*
+	 * The CREATE USER command should not propagate, so we temporarily
+	 * disable DDL propagation.
+	 */
+	SendCommandListToWorkerOutsideTransaction(
+		connection->hostname, connection->port, connection->user,
+		list_make2(
+			"SET LOCAL citus.enable_ddl_propagation TO OFF;",
+			psprintf(
+				"CREATE USER %s SUPERUSER IN ROLE %s",
```

Comment:
Can you explain why this Create User is required? Also add a comment since this is not obvious.

Reply:
Done.